### PR TITLE
Fix issue with reset counters.

### PR DESCRIPTION
**Pull request by n-sandeep**
Check if the counter type (via datafiled info) is available for a table, only then proceed with resetting the counter.

Signed-off-by: Richa Bharti <richa.bharti@intel.com>

### CODE REVIEW

**Review by vsureshkumarp — APPROVED** [review]
Changes looks good to me.

File: stratum/hal/lib/tdi/tdi_sde_counter.cc [hunk 1]
```diff
@@ -57,33 +57,28 @@ ::util::Status TdiSdeWrapper::WriteIndirectCounter(
   if (byte_count.has_value()) {
     tdi_id_t field_id;
     dataFieldInfo = table->tableInfoGet()->dataFieldGet(kCounterBytes);
-    RETURN_IF_NULL(dataFieldInfo);
-    field_id = dataFieldInfo->idGet();
-    RETURN_IF_TDI_ERROR(table_data->setValue(field_id, byte_count.value()));
+    if (dataFieldInfo) {
```
Comment (5abeel) [inline stratum/hal/lib/tdi/tdi_sde_counter.cc:60]: This code is doing the same as old code. Old code returns if null and new code doesnt seem to do anything different. Am I missing something?

File: stratum/hal/lib/tdi/tdi_sde_counter.cc [hunk 2]
```diff
@@ -57,33 +57,28 @@ ::util::Status TdiSdeWrapper::WriteIndirectCounter(
...
   const auto flags = ::tdi::Flags(0);
-  if(byte_count.value() == 0 && packet_count.value() == 0) {
-    LOG(INFO) << "Resetting counters";
-    RETURN_IF_TDI_ERROR(table->clear(
-      *real_session->tdi_session_, *dev_tgt, flags));
-  } else {
-    RETURN_IF_TDI_ERROR(table->entryMod(
-     *real_session->tdi_session_, *dev_tgt, flags, *table_key, *table_data));
-  }
+  RETURN_IF_TDI_ERROR(table->entryMod(
```
Comment (5abeel) [inline stratum/hal/lib/tdi/tdi_sde_counter.cc:80]: This is an API change. 
If I remember correctly, the reset counters needed a `table->clear()` call from SDE side. Has this changed?

File: stratum/hal/lib/tdi/tdi_sde_counter.cc (hunk 1 above)
Comment (vsureshkumarp) [reply stratum/hal/lib/tdi/tdi_sde_counter.cc:60]: There is a difference between Sabeel. Old code flags the error and returns from the processing.
Modified code will continue with further processing for available fields.

File: stratum/hal/lib/tdi/tdi_sde_counter.cc (hunk 2 above)
Comment (vsureshkumarp) [reply stratum/hal/lib/tdi/tdi_sde_counter.cc:80]: The method TdiSdeWrapper::WriteIndirectCounter is meant for resetting the single counter index. It is same as Drop 2 code. Even latest Stratum code also corrected the issue which was there with old code.  For your reference
https://github.com/stratum/stratum/blob/main/stratum/hal/lib/barefoot/bf_sde_wrapper.cc#L2678-L2720

**Review by 5abeel — APPROVED** [review]
looks good

**Pull request merged**